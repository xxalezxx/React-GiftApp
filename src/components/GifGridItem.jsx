import React from "react"
import PropTypes from "prop-types"

export const GifGridItem = ({ id, title, url }) => {
  return (
    <div className="animate__animated animate__bounce animate__delay-2s">
      <img src={url} alt={title}></img>
      <p>{title}</p>
    </div>
  )
}

GifGridItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}

export default GifGridItem
