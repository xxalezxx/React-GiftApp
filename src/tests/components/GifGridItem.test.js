import React from "react"
import { shallow } from "enzyme"
import "@testing-library/jest-dom"

import { GifGridItem } from "../../components/GifGridItem"

describe("GifGridItem.js - Test", () => {
  const title = "something"
  const url = "https://localhost.com"
  const id = "123"

  test("Should show the correct component with all the corrects values", () => {
    const wrapper = shallow(<GifGridItem title={title} url={url} id={id} />)
    expect(wrapper).toMatchSnapshot()
  })

  test("Should have the <p> with the title i", () => {
    const wrapper = shallow(<GifGridItem title={title} url={url} id={id} />)
    const p = wrapper.find("p")
    expect(p.text().trim()).toBe(title)
  })

  test("Should have the same image as url props", () => {
    const wrapper = shallow(<GifGridItem title={title} url={url} id={id} />)
    const img = wrapper.find("img").prop("src")
    expect(img).toBe(url)
  })

  test("Should have animate_something", () => {
    const wrapper = shallow(<GifGridItem title={title} url={url} id={id} />)
    const animation = "animate__animated animate__bounce animate__delay-2s"
    const animationInside = wrapper.find("div").prop("className")
    expect(animation).toBe(animationInside)
  })
})
