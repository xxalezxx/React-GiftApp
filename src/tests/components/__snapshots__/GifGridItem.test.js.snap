// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem.js - Test Should show the correct component with all the corrects values 1`] = `
<div
  className="animate__animated animate__bounce animate__delay-2s"
>
  <img
    alt="something"
    src="https://localhost.com"
  />
  <p>
    something
  </p>
</div>
`;
